### PR TITLE
Don't treat counter fields in outside of tsdb as counters.

### PR DESCRIPTION
**Pull request by martijnvg**
Fields that have the `time_series_metric` attribute set to `counter` in non tsdb indices should use number value source type instead of counter value source type. Essentially not handling these fields as counters at search time.

Relates to #93539

### CODE REVIEW

File: modules/aggregations/src/yamlRestTest/resources/rest-api-spec/test/aggregations/avg_metric.yml [hunk 1]
```diff
@@ -175,3 +175,57 @@ setup:
             the_string_avg:
               avg:
                 field: string_field
+
+---
+"Counter field":
```
Comment (csoulios) [inline modules/aggregations/src/yamlRestTest/resources/rest-api-spec/test/aggregations/avg_metric.yml:180]: So far, all tsdb related tests have been added under `rest-api-spec/test/tsdb` directory. I understand that this is the `average` aggregation, but I think it leaks decisions made for the tsdb object. 

I understand this is a matter of taste, but If was to look for counter related tests, I would be looking in tsdb related directories

File: server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java [hunk 2]
```diff
@@ -1477,7 +1478,8 @@ public NumberFieldType(
             Map<String, String> meta,
             FieldValues<Number> script,
             boolean isDimension,
-            MetricType metricType
+            MetricType metricType,
+            IndexMode indexMode
```
Comment (csoulios) [inline server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java:1482]: I wish there was a better way to implement this without leaking the index mode into so many field mappers.

**Review by csoulios — APPROVED** [review]
LGTM!

Left couple minor comments.

File: server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java (hunk 2 above)
Comment (martijnvg) [reply server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java:1482]: I agree that this is kind of annoying. I don't see another way and this happened in other field mappers as well.

**Pull request merged**